Python/Product/VSInterpreters/Interpreter/InterpreterRegistryService.cs

```diff
@@ -86,6 +98,21 @@ sealed class InterpreterRegistryService : IInterpreterRegistryService, IDisposab
             }
         }
 
+        // Called when a single async interpreter factory provider finishes discovering interpreters
+        private void Provider_AsyncInterpreterDiscoveryCompleted(object sender, EventArgs e) {
+
+            // Since we know how many async providers we have, keep track of how many times this callback is hit.
+            // Once the number of calls == the number of providers, we know all async interpreter discovery is done.
+            _asyncInterpreterDiscoveryCompletedCount++;
```

Comment:
Are these all guaranteed to come on the same thread? It might be worth adding a comment to that effect on `InterpreterDiscoveryCompleted` if so, but if not, this needs to be an atomic increment-and-check.

Reply:
Added a locking mechanism 😄